addon/scope.js

```diff
@@ -0,0 +1,60 @@
+class Scope {
```

Comment:
>We could just do it anyway and hope that our users would never call those methods.

If we're going down that route, I think we should catch anything that won't work and raise an error. I would prefer that they are full aware that it isn't going to work the way they expect.